Lib/asyncio/tasks.py

```diff
@@ -75,6 +76,31 @@ def _set_task_name(task, name):
             set_name(name)
 
 
+@contextlib.contextmanager
+def shield_scope():
+    shield = object()  # just for making sure shields closed in order
+    # if __enter__ and __exit__ called directly
```

Comment:
But who would do that? Let them suffer. `Task().shields` can just be an integer counter.